rskj-core/src/main/java/co/rsk/jsonrpc/JsonRpcIdentifiableMessage.java

```diff
@@ -22,28 +22,30 @@
 /**
  * The basic JSON-RPC request or response. It is required to have an ID.
  *
- * Note that the JSON-RPC 2.0 spec allows using strings as IDs, but our implementation doesn't.
+ * Note that the JSON-RPC 2.0 spec allows using strings as IDs.
  */
 public abstract class JsonRpcIdentifiableMessage extends JsonRpcMessage {
-    private final int id;
+	
+	private static final String ID_BAD_PARAMETER_MSG = "JSON-RPC message id should be a String or a positive number, but was %s.";
 
-    public JsonRpcIdentifiableMessage(JsonRpcVersion version, int id) {
+	private final String id;
+    
+    protected JsonRpcIdentifiableMessage(JsonRpcVersion version, String id) {
         super(version);
-        this.id = requireNonNegative(id);
+        requireNonEmpty(id);
+        this.id = id;
     }
 
     @JsonInclude(JsonInclude.Include.ALWAYS)
-    public int getId() {
+    public String getId() {
```

Comment:
Wouldn't this affect _everyone_ using the ID for something? If users where expecting ints before, this would break their code.

The previous behavior seems a bit weird: if I send an ID as a string ("2") I get the same ID in the response as a string. But if I send an int, I get it as an int. This is weird but according to the spec:

```
An identifier established by the Client that MUST contain a String, Number, or NULL value if included. If it is not included it is assumed to be a notification. The value SHOULD normally not be Null [1] and Numbers SHOULD NOT contain fractional parts [2]

The Server MUST reply with the same value in the Response object if included. This member is used to correlate the context between the two objects.
```

But now it seems that even if I send an integer, the server will reply with a string. I don't think that's wanted.